cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/ArtifactCacheService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.http.CommonNettyHttpServiceBuilder;
+import io.cdap.cdap.internal.app.worker.sidecar.ArtifactLocalizerCleaner;
+import io.cdap.http.NettyHttpService;
+import org.apache.twill.common.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Paths;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Launches an HTTP server for fetching and cache artifacts from a remote CDAP instance.
+ */
+public class ArtifactCacheService extends AbstractIdleService {
```

Comment:
why do we need this to be http service? CDAP on k8s will call end points exposed from this service?

Reply:
I was thinking that we would run the artifact cache service in a separate k8s pod, so requests to fetch artifacts would be http requests to artifact cache service.
Let's discuss further if you think artifact cache service should run locally, just make API calls to fetch artifacts.

Reply:
Ah I see.. If its not serving any endpoints, we dont need to make it run http service.